docs/excel/excel-add-ins-ranges-precedents-dependents.md

```diff
@@ -0,0 +1,97 @@
+---
+title: Work with formula precedents and dependents using the Excel JavaScript API
+description: 'Learn how to use the Excel JavaScript API to retrieve formula precedents and dependents.' 
+ms.date: 05/28/2021
+ms.prod: excel
+localization_priority: Normal
+---
+
+# Get formula precedents and dependents using the Excel JavaScript API
+
+This article provides code samples that retrieve formula precedents and formula dependents using the Excel JavaScript API. For the complete list of properties and methods that the `Range` object supports, see [Range Object (JavaScript API for Excel)](/javascript/api/excel/excel.range).
+
+An Excel formula often refers to other cells. When a cell provides data to a formula, it is known as a formula "precedent". When a cell contains a formula that refers to other cells, it is known as a formula "dependent". To learn more about Excel features related to relationships between cells, see [Display the relationships between formulas and cells](https://support.microsoft.com/office/display-the-relationships-between-formulas-and-cells-a59bef2b-3701-46bf-8ff1-d3518771d507).
+
+## Get formula precedents
+
+With [Range.getDirectPrecedents](/javascript/api/excel/excel.range#getdirectprecedents--), your add-in can locate a formula's direct precedent cells. `Range.getDirectPrecedents` returns a `WorkbookRangeAreas` object. This object contains the addresses of all the precedents in the workbook. It has a separate `RangeAreas` object for each worksheet containing at least one formula precedent. For more information on working with the `RangeAreas` object, see [Work with multiple ranges simultaneously in Excel add-ins](excel-add-ins-multiple-ranges.md).
+
+In the Excel UI, the **Trace Precedents** button draws an arrow from precedent cells to the selected formula. Unlike the Excel UI button, the `getDirectPrecedents` method does not draw arrows.
```

Comment:
I'm not sure people would expect the arrow drawing behavior from a `get` method. What's the motivation to call this out?

Reply:
That said, a screenshot from the Trace Precedents action might helpful in this article. Not only does it visually show what a precedent is, it adds some variety to break up the article.